### PR TITLE
Add liftWithConn to PostgresRaw class

### PR DESCRIPTION
**Pull request by phadej**
This allows to use PostgresRaw with other postgresql-simple compatible
libraries.

### CODE REVIEW

**Comment by phadej** [comment]
In fact, all other methods may be removed from the class; making the dictionary lighter. That might matter or no.

**Comment by 3noch** [comment]
I actually *just* did this locally yesterday. Love it when a plan comes together. However, I just introduced a new class `HasPgConn` to express this separately. It is strictly more general, as you say.

**Comment by phadej** [comment]
Whenever way fits my use case. I can add the `HasPgConn` class as well, though if you have it already you could add your version?

File: backend-db/Rhyolite/Backend/DB/PsqlSimple.hs [hunk 1]
```diff
@@ -127,6 +126,10 @@ class PostgresRaw m where
   default returning :: (m ~ t n, ToRow q, FromRow r, PostgresRaw n, Monad n, MonadTrans t) => Query -> [q] -> m [r]
   returning psql qs = lift $ returning psql qs
 
+  -- | Access raw @postgresql-simple@ connection.
+  liftWithConn :: (Connection -> IO a) -> m a
+  default liftWithConn :: (m ~ t n, PostgresRaw n, Monad n, MonadTrans t) => (Connection -> IO a) -> m a
+  liftWithConn f = lift $ liftWithConn f
```
Comment (3noch) [inline backend-db/Rhyolite/Backend/DB/PsqlSimple.hs]: Why not just expose the `Reader`ness directly with something like `askPgConn`?

**Comment by 3noch** [comment]
I think migrating `PostgresRaw` to something more like `HasPgConn` makes sense, without having to write the instances for a new class.

**Comment by phadej** [comment]
```diff
-class PostgresRaw m where
-  execute :: ToRow q => Query -> q -> m Int64
- ...
- ...
+class MonadIO m => PostgresRaw m where
+  askConn :: m Connection
+  default askConn :: ( m ~ t n , MonadTrans t, Monad n, PostgresRaw n) => m Connection
+  askConn = lift askConn
```

How does this look?

**Comment by 3noch** [comment]
@phadej Looks good to me. @ali-abrar thoughts?

**Comment by ryantrinkle** [comment]
@3noch @phadej I'd prefer to have `HasPgConn` as a separate class, since it makes a much stronger demand on the underlying implementation - i.e. that the functionality *must* be implemented by way of a `Connection` - whereas the existing class only exposes capabilities, i.e. that queries can be run *somehow*.  Other than that, this change looks good to me.

**Comment by 3noch** [comment]
@phadej I'll add my class in this case, as you asked. Thanks for the contribution.

**Pull request closed without merging**